app/views/dashboard/student.haml

```diff
@@ -8,32 +8,26 @@
       %li Your #{interview.application_submission.department.name} interview
       has been scheduled for #{interview.information}.
     .interview_review_link
-      = link_to 'Review your application', interview.application_submission
+      = link_to 'View your application', interview.application_submission
 %h2 Available Applications
 Please select the application which you would like to submit.
 - @positions.each do |department, positions|
   %h3= department.name
   - positions.each do |position|
-    - app_submitted = @application_submissions.try(:keys).try(:include?, position)
-    - if app_submitted
-      // most recent application, if more than one
-      - application = @application_submissions[position].last
-      - should_show = application.pending? || should_show_denied_applications?
-    - if app_submitted && should_show
-      %li= link_to "Review your application for #{position.name}", application
-    - if position.application_template.try(:active?)
-      - if app_submitted
-        - if should_allow_resubmission?
-          -if application.reviewed
+    // most recent application, if more than one
+    - if application = @application_submissions.try(:[], position)&.last
+      - if application.pending? || should_show_denied_applications?
+        %li= link_to "View your submitted application for #{position.name}", application
+        - if position.application_template.try(:active?) && should_allow_resubmission?
+          - if application.reviewed?
             %li= link_to "Resubmit application for #{position.name}",
               application_path(position.application_template)
           - else
-            Your application has not been reviewed yet.
+            Your application has not yet been reviewed by staff
         - else
           You are not allowed to resubmit this application.
-      -else
-        %li= link_to "Submit application for #{position.name}",
-          application_path(position.application_template)
     - else
-      %li
-        = position.not_hiring
+      - if position.application_template.try(:active?)
+        %li= link_to "Submit application for #{position.name}", application_path(position.application_template)
+      - else
+        %li= position.not_hiring
```

Comment:
This could be collapsed one `if`-level:

```haml
    - elsif position.application_template.try(:active?)
      %li= link_to "Submit application for #{position.name}", application_path(position.application_template)
    - else
      %li= position.not_hiring
```